src/commands/test.js

```diff
@@ -90,3 +141,109 @@ function getPlatform(logs) {
 
   throw new Error(`Unsupported platform: ${type} ${arch} ${majorVersion}`)
 }
+
+function runDocker(coverage_opt, datasource, version_opt, latest_version, print) {
```

Comment:
Looking at this function I think you can improve it by using the built-in `toolbox` from `gluegun` instead of `fs` and `child_process`.

It really shouldn't change behavior, the nice thing is that everything is promisified 🙂 

Take a look at this function for instance: https://github.com/graphprotocol/graph-cli/blob/master/src/commands/init.js#L556

Reply:
The point of using `child_process.spawn()` with `{stdio: inherit}` options was to have the docker output in real time, as well as the matchstick logs colors, as if it is ran locally. I will look into it though.

Reply:
Looking through the toolbox docs, `toolbox.system.run()` is `a promise wrapped around node.js child-process.exec api call`, so unfortunately it won't work for us in this case. But I think we can safely replace `fs` with `toolbox.filesystem`